### PR TITLE
Rename original resolv.conf instead of rewriting

### PR DESCRIPTION
**Pull request by kayrus**
supersedes #17
@HeikoSchlittermann can you check this PR and let me know if it suits your expectations?

### CODE REVIEW

**Comment by HeikoSchlittermann** [comment]
Kayrus, thank you for the notification and sorry for the delayed response, I was on a short vacation trip. I'll review your changes.

File: pkg/config.go [hunk 1]
```diff
@@ -98,6 +98,10 @@ func saveCookies(c *http.Client, u *url.URL, config *Config) error {
 }
 
 func parseResolvConf(config *Config) {
+	if config.resolvConf == nil {
```
Comment (HeikoSchlittermann) [inline pkg/config.go:101]: Good move :) If it is `nil`, there was no `resolv.conf`, if it is just empty, the _existing_ `resolv.conf` was empty.

**Review by HeikoSchlittermann — APPROVED** [review]
Yes, it should match my expectations. I suggested some changes. Beside a type I'd rely on the error messages produced by Go itself, the're generally verbose enough. But it's a matter of stile and up to you :)

I didn't do a run-test with your changes, I first have to rebase other changes I did.

File: pkg/resolv_all.go [hunk 2]
```diff
@@ -30,18 +36,40 @@ func configureDNS(config *Config) error {
...
-	if err := ioutil.WriteFile(resolvPath, dns.Bytes(), 0644); err != nil {
+
+      // default "/etc/resolv.conf" permissions
+	var perm os.FileMode = 0644
+	if config.resolvConf != nil {
+		info, err := os.Stat(resolvPath)
+		if err != nil {
+			return err
+		}
+		// reuse the original "/etc/resolv.com" permissions
```
Comment (kayrus) [inline pkg/resolv_all.go:47]:
```suggestion
		// reuse the original "/etc/resolv.conf" permissions
```

**Pull request merged**